src/ashell/shell-state.c

```diff
@@ -594,8 +594,10 @@ int32_t ashell_set_bootcfg(char *buf)
         return RET_ERROR;
     }
 
-    ssize_t written = fs_write(file, buf, strlen(buf));
-    if (written == 0) {
+    ssize_t written = fs_write(file, __TIMESTAMP__, strlen(__TIMESTAMP__));
```

Comment:
Good find, I didn't know about `__TIMESTAMP__`. However, it looks like this is the last modification time of the source file. So I think you would be better off with `__TIME__` which is apparently the current time when the preprocessor is running.

Also, I'd suggest you use the macro once at the top of the file like this:

```
const char *BUILD_TIMESTAMP = __DATE__ " " __TIME__ "\n";
```

This is just because gcc docs aren't clear on whether this is set once per source file or changes during the course of processing the source file. If the latter, then it's possible for you to hit just right where it would roll over to the next second between when you use it here and down below. Then you'd have a bug where it never matches and you'd erase boot.cfg every time they create it I think.